server/common/utils/helpers.js

```diff
@@ -401,6 +401,38 @@ const fireAndForget = promise => promise.catch(err => console.error(err));
 
 const returnPayload = promise => payload => promise.then(() => payload);
 
+const responseWithListDetails = (list, userId) => cohort => {
+  const {
+    isArchived,
+    items: listItems,
+    memberIds,
+    ownerIds,
+    viewersIds: viewersCollection
+  } = list;
+  const activeItems = listItems.filter(item => !item.isArchived);
+  const items = responseWithItems(userId, activeItems);
+  const cohortMembers = cohort ? cohort.memberIds : [];
+
+  const members = responseWithListMembers(
+    viewersCollection,
+    memberIds,
+    ownerIds,
+    cohortMembers
+  );
+
+  return {
+    ...responseWithList(list, userId),
+    cohortId: cohort ? cohort.cohortId : null,
+    cohortName: cohort ? cohort.name : null,
+    isArchived,
+    isGuest: !cohort || (cohort && !isMember(cohort, userId)),
+    isMember: isMember(list, userId),
+    isOwner: isOwner(list, userId),
+    items: _keyBy(items, '_id'),
+    members: _keyBy(members, '_id')
```

Comment:
just curious why keyby here? 
we were always using keyby on front end before dispatching data to the redux

Reply:
We did it in action on fetch result. So I should implement this again in `listEventsController` helpers function. I didn't want to make this method any more complicated. So I send ready list object from backend